src/function/logical/nullish.js

```diff
@@ -0,0 +1,113 @@
+import { DimensionError } from '../../error/DimensionError.js'
+
+const name = 'nullish'
+const dependencies = ['typed', 'matrix']
+
+export const createNullish = /* #__PURE__ */ factory(
+  name,
+  dependencies,
+  ({ typed, matrix }) => {
+    const matAlgo03xDSf = createMatAlgo03xDSf({ typed })
+    const matAlgo14xDs = createMatAlgo14xDs({ typed })
+    const matAlgo13xDD = createMatAlgo13xDD({ typed })
+
+    /**
+     * Nullish coalescing operator (??). Returns the right-hand side operand
+     * when the left-hand side operand is null or undefined, and otherwise
+     * returns the left-hand side operand.
+     *
+     * For matrices, the function is evaluated element wise.
+     *
+     * Syntax:
+     *
+     *    math.nullish(x, y)
+     *
+     * Examples:
+     *
+     *    math.nullish(null, 42)        // returns 42
+     *    math.nullish(undefined, 42)   // returns 42
+     *    math.nullish(0, 42)           // returns 0
+     *    math.nullish(false, 42)       // returns false
+     *    math.nullish('', 42)          // returns ''
+     *
+     *    // Object property access with fallback
+     *    const obj = {foo: 7, bar: 3}
+     *    math.nullish(obj.baz, 0)      // returns 0 (undefined ?? 0)
+     *
+     * See also:
+     *
+     *    and, or, not
+     *
+     * @param  {*} x First value to check
+     * @param  {*} y Fallback value
+     * @return {*} Returns y when x is null or undefined, otherwise returns x
+     */
+
+    // Helper function to check if a value is nullish
+    const isNullish = (x) => x == null || x === undefined
+
+    return typed(
+      name,
+      {
+        // Scalar handlers
+        'number|bigint|Complex|BigNumber|Fraction|Unit|string|boolean|SparseMatrix, any': (x, _y) => x,
+        'null, any': (_x, y) => y,
+        'undefined, any': (_x, y) => y,
+
+        // Matrix-aware overloads and array bridging (no broadcasting)
+        'DenseMatrix, any': typed.referToSelf(self => (x, y) =>
+          // element-wise evaluation over dense with scalar/array-like
+          matAlgo14xDs(x, y, self, false)
+        ),
+
+        'SparseMatrix, DenseMatrix': (x, y) => {
+          const xs = x.size()
+          const ys = y.size()
+          if (xs.length !== ys.length) {
+            throw new DimensionError(xs.length, ys.length)
+          }
+          for (let i = 0; i < xs.length; i++) {
+            if (xs[i] !== ys[i]) {
+              throw new DimensionError(xs, ys)
+            }
+          }
+          return x
+        },
+        'SparseMatrix, Array': (x, y) => {
+          const ym = matrix(y)
+          const xs = x.size()
+          const ys = ym.size()
+          if (xs.length !== ys.length) {
+            throw new DimensionError(xs.length, ys.length)
+          }
+          for (let i = 0; i < xs.length; i++) {
+            if (xs[i] !== ys[i]) {
+              throw new DimensionError(xs, ys)
+            }
+          }
+          return x
+        },
+
```

Comment:
These two functions do the same thing: check the sizes match and then return the left operand. So there should only be one function called twice, not repeated code. Thanks!